src/core/services.py

```diff
@@ -655,7 +655,7 @@ def get_directory_listing(
     else:
         directory = path
 
-    normalized_path = pathlib.Path(directory)
+    normalized_path = pathlib.Path(os.path.normpath(directory))
```

Comment:
if it do not resolve the issue, may need something like this as well, to catch `../../../xxx` type of paths.
```py
    normalized_path = pathlib.Path(os.path.normpath(directory))
    if not str(normalized_path).startswith(directory):
        raise Exception("fishy path")
```
and remove the else branch above.

Reply:
Thanks for the suggestion.  I'll implement something like this if CodeQL does not get happy with this. 😄